src/mpid/ch4/src/ch4_init.c

```diff
@@ -664,9 +664,12 @@ void *MPID_Alloc_mem(MPI_Aint size, MPIR_Info * info_ptr)
              * process is bound to the corresponding device; allocate
              * memory and bind it to device. */
             assert(mem_gid != MPIR_HWTOPO_GID_ROOT);
-            real_buf =
-                MPL_mmap(NULL, size + alignment, PROT_READ | PROT_WRITE, MAP_ANON | MAP_PRIVATE, -1,
-                         0, MPL_MEM_USER);
+#ifdef MAP_ANON
+            real_buf = MPL_mmap(NULL, size + alignment, PROT_READ | PROT_WRITE,
+                                MAP_ANON | MAP_PRIVATE, -1, 0, MPL_MEM_USER);
+#else
+            real_buf = MPL_malloc(size + alignment, MPL_MEM_USER);
+#endif
```

Comment:
Honestly I think we can just use malloc in all cases. I don't what benefit there is to using mmap.

Reply:
I think `mmap` always allocates a new page which may be more suitable for pinning.

Reply:
Hm, yes that is document behavior for Linux. Fair enough I suppose.